### PR TITLE
Migrate from Jest to Vitest, update ESLint config and upgrade Execa

### PR DESCRIPTION
**Pull request by ghengeveld**
All this to allow us to use AbortSignal with Execa...

### CODE REVIEW

**Review by tmeasday — APPROVED** [review]
Seems fine although I didn't review it all. Is there any thing in there I should look at?

**Comment by ghengeveld** [comment]
@tmeasday The tests for `getParentCommits` use `process.chdir` which isn't supported in Vitest without passing `--no-threads`, so I ended up configuring it to run separately. I think that might be worth checking out but otherwise I feel pretty good about this change.

File: node-src/git/getParentCommits.test.ts [hunk 1]
```diff
@@ -76,17 +72,18 @@ interface Repository {
...
       const runGit = makeRunGit(dirname);
       const commitMap = await generateGitRepository(runGit, descriptions[key]);
       repositories[key] = { dirname, runGit, commitMap };
     })
-  )
-);
+  );
+});
 
+// Skipped because vitest doesn't allo `chdir` in workers
```
Comment (tmeasday) [inline node-src/git/getParentCommits.test.ts:86]: Not any more?

**Review by tmeasday — APPROVED** [review]
Seems reasonable

**Pull request merged**